### PR TITLE
feature: Apache Airflow integration for SageMaker Processing Jobs

### PR DESCRIPTION
**Pull request by guptamyr**
*Issue #, if available:*

*Description of changes:*
This change enables an Apache Airflow operator for SageMaker Processing Jobs through a processing_config that can be passed to an Airflow operator. 
*Testing done:*
Added a unit test for this config. Also tested this with local changes to the Airflow repository that add the Operator and was able to create/run a DAG for ProcessingJobs similar to the examples [here](https://github.com/aws/sagemaker-python-sdk/blob/5d74516f17de89dfc2537b9516b38de53f7273ed/doc/workflows/airflow/using_workflow.rst).

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/processing.py [hunk 1]
```diff
@@ -739,6 +739,66 @@ def stop(self):
...
+            container_arguments:
+            container_entrypoint:
+            image_uri:
+        """
+        config = {}
+        if container_arguments is not None:
+            config["ContainerArguments"] = container_arguments
+        if container_entrypoint is not None:
+            config["ContainerEntrypoint"] = container_entrypoint
+        config["ImageUri"] = image_uri
```
[the remark this answers is not in the document]
Comment (guptamyr) [reply src/sagemaker/processing.py:755]: Fixed in the next revision.

File: src/sagemaker/processing.py [hunk 2]
```diff
@@ -739,6 +739,66 @@ def stop(self):
...
+    def _prepare_output_config(kms_key_id, outputs):
+        """
+        Args:
+            kms_key_id:
+            outputs:
+        """
+        config = {}
+        if kms_key_id is not None:
+            config["KmsKeyId"] = kms_key_id
+        config["Outputs"] = outputs
```
[the remark this answers is not in the document]
Comment (guptamyr) [reply src/sagemaker/processing.py:768]: Fixed in the next revision.

File: src/sagemaker/processing.py [hunk 3]
```diff
@@ -739,6 +739,66 @@ def stop(self):
...
+
+    @staticmethod
+    def _prepare_stopping_condition(max_runtime_in_seconds):
+        """
+        Args:
+            max_runtime_in_seconds
+        """
+        stopping_condition = {}
+        stopping_condition["MaxRuntimeInSeconds"] = max_runtime_in_seconds
+        return stopping_condition
```
[the remark this answers is not in the document]
Comment (guptamyr) [reply src/sagemaker/processing.py:800]: Fixed in the next revision.

File: src/sagemaker/processing.py [hunk 4]
```diff
@@ -739,6 +739,66 @@ def stop(self):
         """Stops the processing job."""
         self.sagemaker_session.stop_processing_job(self.name)
 
+    @staticmethod
+    def _prepare_app_specification(container_arguments, container_entrypoint, image_uri):
```
[the remark this answers is not in the document]
Comment (guptamyr) [reply src/sagemaker/processing.py:743]: Removed the underscore in the next revision.

File: src/sagemaker/processing.py [hunk 5]
```diff
@@ -739,6 +739,66 @@ def stop(self):
...
+        if container_arguments is not None:
+            config["ContainerArguments"] = container_arguments
+        if container_entrypoint is not None:
+            config["ContainerEntrypoint"] = container_entrypoint
+        config["ImageUri"] = image_uri
+        return config
+
+    @staticmethod
+    def _prepare_output_config(kms_key_id, outputs):
+        """
```
[the remark this answers is not in the document]
Comment (guptamyr) [reply src/sagemaker/processing.py:760]: Fixed the underscore and added summary lines, descriptions of args and type for all of these functions.

File: src/sagemaker/processing.py [hunk 6]
```diff
@@ -739,6 +739,66 @@ def stop(self):
...
+        Args:
+            kms_key_id:
```
[the remark this answers is not in the document]
Comment (guptamyr) [reply src/sagemaker/processing.py:762]: Fixed the underscore and added summary lines, descriptions of args and type for all of these functions.

File: src/sagemaker/workflow/airflow.py [hunk 7]
```diff
@@ -1065,3 +1065,99 @@ def deploy_config_from_estimator(
...
+        processor._current_job_name = job_name
+    else:
+        base_name = processor.base_job_name
+        processor._current_job_name = (
+            utils.name_from_base(base_name)
+            if base_name is not None
+            else utils.base_name_from_image(processor.image_uri)
+        )
+
+    input_dict = []
```
[the remark this answers is not in the document]
Comment (guptamyr) [reply src/sagemaker/workflow/airflow.py:1117]: Fixed in the next revision.

File: src/sagemaker/workflow/airflow.py [hunk 8]
```diff
@@ -1065,3 +1065,99 @@ def deploy_config_from_estimator(
...
+    for i in inputs:
+        input_dict.append(i._to_request_dict())
```
[the remark this answers is not in the document]
Comment (guptamyr) [reply src/sagemaker/workflow/airflow.py:1119]: Fixed in the next revision.

File: src/sagemaker/workflow/airflow.py [hunk 9]
```diff
@@ -1065,3 +1065,99 @@ def deploy_config_from_estimator(
...
+    config = {"ProcessingJobName": processor._current_job_name, "ProcessingInputs": input_dict}
+
+    output_dict = []
+    for output in outputs:
+        output_dict.append(output._to_request_dict())
```
[the remark this answers is not in the document]
Comment (guptamyr) [reply src/sagemaker/workflow/airflow.py:1124]: Added a helper function in the next revision.

File: src/sagemaker/processing.py [hunk 10]
```diff
@@ -739,6 +739,94 @@ def stop(self):
...
+        """
+        processing_resources = {}
+        cluster_config = {}
+        if volume_kms_key_id is not None:
+            cluster_config["VolumeKmsKeyId"] = volume_kms_key_id
+        cluster_config["InstanceCount"] = instance_count
+        cluster_config["InstanceType"] = instance_type
+        cluster_config["VolumeSizeInGB"] = volume_size_in_gb
+        processing_resources["ClusterConfig"] = cluster_config
+        return processing_resources
```
Comment (metrizable) [inline src/sagemaker/processing.py:815]: nit-style: initialize dict with values?
```
cluster_config = {
    "InstanceCount": instance_count,
    ...
}
...
return {"ClusterConfig": cluster_config}
```
}

File: src/sagemaker/processing.py [hunk 11]
```diff
@@ -739,6 +739,94 @@ def stop(self):
...
+        config = {"ImageUri": image_uri}
+        if container_arguments is not None:
+            config["ContainerArguments"] = container_arguments
+        if container_entrypoint is not None:
+            config["ContainerEntrypoint"] = container_entrypoint
+        return config
+
+    @staticmethod
+    def prepare_output_config(kms_key_id, outputs):
+        """Prepares a dict that represents a ProcessingOutputConfig,
```
Comment (metrizable) [inline src/sagemaker/processing.py:766]: nit-[pep257](https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings): single line summary, additional information after blank line in body.

File: src/sagemaker/processing.py [hunk 12]
```diff
@@ -739,6 +739,94 @@ def stop(self):
         """Stops the processing job."""
         self.sagemaker_session.stop_processing_job(self.name)
 
+    @staticmethod
+    def prepare_app_specification(container_arguments, container_entrypoint, image_uri):
+        """Prepares a dict that represents a ProcessingJob's AppSpecification
```
Comment (metrizable) [inline src/sagemaker/processing.py:744]: nit-pep257

File: src/sagemaker/processing.py (hunk 11 above)
Comment (guptamyr) [reply src/sagemaker/processing.py]: Fixed in the next revision. (moved the extra information to describe what the function returns.

File: src/sagemaker/processing.py (hunk 10 above)
Comment (guptamyr) [reply src/sagemaker/processing.py]: Fixed in the next revision.

File: src/sagemaker/processing.py (hunk 12 above)
Comment (guptamyr) [reply src/sagemaker/processing.py]: Fixed in the next revision. (moved the extra information to describe what the function returns.)

File: src/sagemaker/processing.py [hunk 13]
```diff
@@ -740,6 +740,93 @@ def stop(self):
...
+                that Amazon SageMaker uses to encrypt data on the storage
+                volume attached to the ML compute instance(s) that run the processing job.
+            volume_size_in_gb (int): The size of the ML storage volume in gigabytes
+                that you want to provision. You must specify sufficient
+                ML storage for your scenario.
+
+        Returns:
+            dict: Represents ProcessingResources which identifies the resources,
+            ML compute instances, and ML storage volumes to deploy
+            for a processing job.
```
Comment (metrizable) [inline src/sagemaker/processing.py:805]: additional lines should be indented as with args

```suggestion
            dict: Represents ProcessingResources which identifies the resources,
                ML compute instances, and ML storage volumes to deploy
                for a processing job.
```

File: src/sagemaker/workflow/airflow.py [hunk 14]
```diff
@@ -1070,3 +1070,108 @@ def deploy_config_from_estimator(
...
+        container_arguments ([str]): The arguments for a container used to run a processing job.
+        container_entrypoint ([str]): The entrypoint for a container used to run a processing job.
+        kms_key_id (str): The AWS Key Management Service (AWS KMS) key that Amazon SageMaker
+            uses to encrypt the processing job output. KmsKeyId can be an ID of a KMS key,
+            ARN of a KMS key, alias of a KMS key, or alias of a KMS key.
+            The KmsKeyId is applied to all outputs.
+
+    Returns:
+        dict: Processing config that can be directly used by
+        SageMakerProcessingOperator in Airflow.
```
Comment (metrizable) [inline src/sagemaker/workflow/airflow.py:1110]: additional lines should be indented

```suggestion
        dict: Processing config that can be directly used by
            SageMakerProcessingOperator in Airflow.
```

**Pull request merged**